jujugui/static/gui/src/app/views/topology/panzoom.js

```diff
@@ -208,16 +208,30 @@ YUI.add('juju-topology-panzoom', function(Y) {
       @return {undefined} Side effects only.
     */
     panToPoint: function(evt) {
-      var point = evt.point,
+      const point = evt.point,
           topo = this.get('component'),
           scale = topo.get('scale'),
           size = [window.innerWidth, window.innerHeight];
-      this.rescale({
-        scale: scale,
-        translate: point.map(function(d, i) {
-          return ((0 - d) * scale) + (size[i] / 2);
-        })
-      });
+      const offset = topo.zoom.translate();
+      const screenWidth = size[0];
+      const screenHeight = size[1];
+      // Translate the points to values that can be compared with the current
+      // screen area.
+      const pointX = (point[0] * scale) + offset[0];
+      const pointY = (point[1] * scale) + offset[1];
+      // Find out if the points we have are within the screen rectangle.
+      const withinScreen = pointX > 0 && pointX < screenWidth && pointY > 0 &&
+        pointY < screenHeight;
+      // If the object is not within the screen the move the screen so it is
+      // visible.
+      if (!withinScreen) {
+        this.rescale({
+          scale: scale,
+          translate: point.map(function(d, i) {
```

Comment:
Mixing new and old (arrowfuncs, let/const vs var) drives me up the wall.

That said, you're right 👍